desktop/core/src/desktop/js/apps/storageBrowser/StorageBrowserPage/StorageBrowserTable/StorageBrowserTable.tsx

```diff
@@ -61,12 +76,50 @@ const StorageBrowserTable: React.FC<StorageBrowserTableProps> = ({
 
   const { t } = i18nReact.useTranslation();
 
+  const onColumnTitleClicked = (columnClicked: string) => {
+    if (columnClicked === sortByColumn) {
+      switch (sortOrder) {
+        case SortEnum.NONE:
+          onSortOrderChange(SortEnum.ASC);
+          break;
+        case SortEnum.ASC:
+          onSortOrderChange(SortEnum.DSC);
+          break;
+        default:
+          onSortOrderChange(SortEnum.NONE);
+      }
+    } else {
+      onSortByColumnChange(columnClicked);
+      onSortOrderChange(SortEnum.ASC);
+    }
+  };
+
   const getColumns = file => {
     const columns: ColumnProps<unknown>[] = [];
     for (const [key] of Object.entries(file)) {
       const column: ColumnProps<unknown> = {
-        dataIndex: `${key}`,
-        title: t(`${key}`.charAt(0).toUpperCase() + `${key}`.slice(1)),
+        dataIndex: key,
+        title: (
+          <div
+            className="hue-storage-browser__table-column-header"
+            onClick={() => onColumnTitleClicked(key)}
+          >
+            <div className="hue-storage-browser__table-column-title">
+              {key === 'mtime' ? t('Last Updated') : t(key)}
+            </div>
+            {key === sortByColumn ? (
+              sortOrder === SortEnum.DSC ? (
+                <SortDescending />
+              ) : sortOrder === SortEnum.ASC ? (
+                <SortAscending />
+              ) : (
+                <></>
```

Comment:
can we just return null here or will that fail?

Reply:
It requires an expression to be sent so I added this. Otherwise it shows a parsing error